### PR TITLE
 Clean up code to build with -Wall -Werror -Wextra

### LINKED ISSUE #29: Poor quality code
This fail horribly with `-Wall -Werror -Wextra"` added, this needs to be fixed.

### PR DESCRIPTION
**Pull request by mariogrip**
This cleans up the code to a stricter and better quality code, this should also fix some of the random crashes that I have seen, these crashes happen when history service hits end of a function without the function returning anything. This would have been completely avoided if -Wall -Werror -Wextra has been enabled. 

Fixes: https://github.com/ubports/history-service/issues/29

### CODE REVIEW

File: plugins/sqlite/sqlitedatabase.cpp [hunk 1]
```diff
@@ -37,21 +37,21 @@ Q_DECLARE_OPAQUE_POINTER(sqlite3*)
...
 {
     QString arg1((const char*)sqlite3_value_text(argv[0]));
     QString arg2((const char*)sqlite3_value_text(argv[1]));
     sqlite3_result_int(context, (int)History::PhoneUtils::compareNormalizedPhoneNumbers(arg1, arg2));
 }
 
-void normalizeId(sqlite3_context *context, int argc, sqlite3_value **argv)
+void normalizeId(sqlite3_context *context, int /* argc */, sqlite3_value **argv)
 {
     QString accountId((const char*)sqlite3_value_text(argv[0]));
```
Comment (mardy) [inline plugins/sqlite/sqlitedatabase.cpp:62]: Might be good to add a check on `argc` being >= 2

**Review by mardy — APPROVED** [review]

File: plugins/sqlite/sqlitedatabase.cpp (hunk 1 above)
Comment (mariogrip) [reply plugins/sqlite/sqlitedatabase.cpp:62]: I agree, let me fix that

Comment (mariogrip) [reply plugins/sqlite/sqlitedatabase.cpp:62]: That should do :)

**Review by dobey — APPROVED** [review]

**Comment by UniversalSuperBox** [comment]
@lduboeuf, have you noticed any behavior changes inside history-service after these changes?

**Comment by lduboeuf** [comment]
> @lduboeuf, have you noticed any behavior changes inside history-service after these changes?

no i don't see any or if any they may be not be visible

**Pull request merged**